### PR TITLE
Feature/621 UI home

### PR DESCRIPTION
**Pull request by Rebal67**

### CODE REVIEW

File: src/utils/theme.ts [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+        grey6: '#F4F4F4',
     },
 };
 
+export type AppTheme = typeof theme;
+
+export const useAppTheme = () => useTheme() as AppTheme;
+
 export const customColors = {
     disabledButtonTextColor: 'darkgrey',
```
Comment (theblockstalk) [inline src/utils/theme.ts:39]: can these become part of our theme instead?

File: src/utils/theme.ts [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
 
 // https://callstack.github.io/react-native-paper/4.0/theming.html
-const theme: Theme = {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const theme = {
     ...DefaultTheme,
     dark: false, // as we are using light theme this should be false
     colors: {
         ...DefaultTheme.colors,
         primary: settings.config.theme.primaryColor,
```
Comment (theblockstalk) [inline src/utils/theme.ts:13]: should this match the colour pallet and structure @MilicaSequeira has defined? maybe new task to do this?
image

File: src/components/TNavigationButton.tsx [hunk 3]
```diff
@@ -14,46 +14,44 @@ export type NavigationButtonProps = {
...
-                </View>
+            {props.icon && typeof props.icon === 'object' && (
+                <View style={{ alignSelf: getAlignmentBasenOnDescription() }}>{props.icon}</View>
             )}
+
+            <View style={{ ...styles.titleContainer }}>
+                <TP
+                    style={{
+                        fontWeight: `${props?.disabled === false ? 'bold' : '100'}`,
+                        color: `${props?.disabled === false ? 'black' : 'gray'}`,
```
Comment (theblockstalk) [inline src/components/TNavigationButton.tsx:36]: should use theme here

File: src/components/atoms/Tbutton.tsx [hunk 4]
```diff
@@ -1,39 +1,109 @@
...
 }
 
 export function TButtonContained(props: ButtonProps) {
+    const theme = useAppTheme();
+    const color = theme.colors.white;
+    const style = {
+        backgroundColor: props.disabled ? theme.colors.primary2 : getColorBasedOnTheme(props.theme),
+    };
+    const shadowStyle = {
+        shadowColor: '#000',
```
Comment (theblockstalk) [inline src/components/atoms/Tbutton.tsx:65]: use theme here?

File: src/navigation/Drawer.tsx [hunk 5]
```diff
@@ -21,14 +21,14 @@ export default function DrawerNavigation() {
     const theme = useTheme();
     const defaultScreenOptions: DrawerNavigationOptions = {
         headerStyle: {
-            backgroundColor: theme.colors.primary,
+            shadowColor: '#4D4D4D',
```
Comment (theblockstalk) [inline src/navigation/Drawer.tsx:24]: theme colour here? a grey?

File: src/navigation/Root.tsx [hunk 6]
```diff
@@ -66,14 +67,15 @@ export default function RootNavigation() {
             background: theme.colors.background,
         },
     };
+
     const defaultScreenOptions: NativeStackNavigationOptions = {
         headerStyle: {
             // backgroundColor: theme.colors.,
```
Comment (theblockstalk) [inline src/navigation/Root.tsx:73]: remove comment

**Review by theblockstalk — COMMENTED** [review]
It would be nice if someone can run the app on another device, and check if no errors.

File: src/utils/theme.ts (hunk 2 above)
Comment (Rebal67) [reply src/utils/theme.ts:13]: this is using that

File: src/utils/theme.ts (hunk 1 above)
Comment (Rebal67) [reply src/utils/theme.ts:39]: can you be more specific

**Pull request merged**